userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1522,7 +1522,7 @@ int32_t scap_bpf_load(
 
 		if(online_cpu >= handle->m_dev_set.m_ndevs)
 		{
-			snprintf(handle->m_lasterr, SCAP_LASTERR_SIZE, "processors online: %d, expected: %d", online_cpu, handle->m_dev_set.m_ndevs);
+			snprintf(handle->m_lasterr, SCAP_LASTERR_SIZE, "too many online processors: %d, expected: %d", online_cpu, handle->m_dev_set.m_ndevs);
```

Comment:
Distinguish this error from the below one (there is another check at the end of the loop).

Reply:
The other check (L1565) was the one triggering the error. Let's make an example:
we were looping on num_cpus == num_ndevs; let's say we had 3/4 online CPUs, like this: 1 0 1 1.

Basically, our loop would loop 3 times (online CPUs -> remember that num_cpus == num_ndevs); BUT the second element would be skipped because offline. 
At the end of the for loop, we had counted 2 online CPUs instead of 3.

Reply:
makes a lot of sense, great catch! I'm missing just one piece of the puzzle why do we skip the check on the first CPU? :point_down:  

```c
		if(j > 0)
		{
			char filename[SCAP_MAX_PATH_SIZE];
			int online;
			FILE *fp;
                        ...........
```

Reply:
Because first CPU has not `online` flag; it cannot be disabled!

Reply:
uh right, thank you!